tls/s2n_record_read.c

```diff
@@ -38,11 +38,18 @@ int s2n_sslv2_record_header_parse(
 
     S2N_ERROR_IF(s2n_stuffer_data_available(in) < S2N_TLS_RECORD_HEADER_LENGTH, S2N_ERR_BAD_MESSAGE);
 
+    /* The first bit of the length must be set to indicate SSLv2,
+     * so the raw fragment length must be adjusted.
+     * Revert the change afterwards in case we need to re-parse the header.
+     */
+    in->blob.data[0] ^= S2N_TLS_SSLV2_HEADER_FLAG;
     POSIX_GUARD(s2n_stuffer_read_uint16(in, fragment_length));
+    in->blob.data[0] |= S2N_TLS_SSLV2_HEADER_FLAG;
```

Comment:
Why not mask off the flag in the `fragment_length` variable rather than modifying the blob? I guess we do have the `s2n_stuffer_data_available` check above, but I'd prefer to just not modify `in` at all in case things get refactored at some later point.

Reply:
I was doing that initially, but it's not as simple as "mask off" because the length is two bytes in network order. Modifying it before we read it seemed clearer, and we'd only not modify it back on a fatal parsing error.

Let me switch back to not modifying it and see if you like that better or worse.